libsupport/include/katana/OpaqueID.h

```diff
@@ -197,7 +198,9 @@ struct OpaqueIDLinear : public OpaqueIDOrderedWithValue<_IDType, _Value> {
 public:
   using OpaqueIDOrderedWithValue<_IDType, _Value>::OpaqueIDOrderedWithValue;
 
-  using DifferenceType = std::make_signed_t<_Value>;
+  static_assert(sizeof(_Value) <= sizeof(std::ptrdiff_t), "The ID is too large to safely use ptrdiff_t");
```

Comment:
Use ptrdiff_t for what? Probably better to say "OpaqueIDLinear only supports values up to the size of ptrdiff_t"